src/server/routes/editor.js

```diff
@@ -368,6 +369,93 @@ router.get('/:id/revisions/revisions', async (req, res, next) => {
 	}
 });
 
+async function getOrderedCollections(from, size, req) {
+	const {UserCollection, UserCollectionCollaborator} = req.app.locals.orm;
+
+	const otherCollections = await new UserCollectionCollaborator({editorId: req.params.id}).fetchAll();
+	const otherCollectionsJSON = otherCollections ? otherCollections.toJSON() : [];
+	const collectionIds = otherCollectionsJSON.map((collection) => collection.collectionId);
+
+	const isThisCurrentUser = req.user && parseInt(req.params.id, 10) === parseInt(req.user.id, 10);
+
+	const allCollections = await new UserCollection()
+		.query((qb) => {
+			if (!isThisCurrentUser) {
+				qb.where('public', true);
+			}
+			qb.whereIn('id', collectionIds)
+				.orWhere('owner_id', req.params.id);
+			qb.orderBy('name');
+		}).fetchPage({
+			limit: size,
+			offset: from,
+			withRelated: [
+				'items'
+			]
+		});
```

Comment:
I think this can done in one DB query, instead of two here. I will have a look once.

Reply:
Also I am thinking of adding `entity_type` in params. That way collections of a particular type can be fetched. 
<br>
This route then can be used in `Add Entity to Collection` in the search page.
`editor/:id/collections/collections?type=edition` this will return a JSON object with list of editon-collection for the user

Reply:
OK, let me know when I should review, after you've looked into doing a single query.


I think the addition of `entity_type` (or just `type`) could be useful indeed.